api/provider/aws/environment_update.go

```diff
@@ -33,6 +39,93 @@ func (e *EnvironmentProvider) Update(req models.UpdateEnvironmentRequest) error
 		}
 	}
 
+	if req.Links != nil {
+		sourceEnvSG, err := readSG(e.AWS.EC2, getEnvironmentSGName(fqEnvironmentID))
+		if err != nil {
+			return err
+		}
+
+		// check which environments are currently linked
+		currentTags, err := e.TagStore.SelectByTypeAndID("environment", environmentID)
+		if err != nil {
+			return err
+		}
+
+		currentLinkedEnvIDs := []string{}
+		for _, tag := range currentTags.WithKey("link") {
+			currentLinkedEnvIDs = strings.Split(tag.Value, ",")
+		}
+
+		checkForAWSError := func(err error) error {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "DoesNotExist" {
+				log.Printf("[WARN] skipping environment link/unlink since security group '%s' does not exist\n", getEnvironmentSGName(fqEnvironmentID))
+				return nil
+			}
+
+			return err
+		}
+
+		contains := func(value string, values []string) bool {
+			for _, v := range values {
+				if v == value {
+					return true
+				}
+			}
+
+			return false
+		}
+
+		// add - check if current link state contains links in request
+		for _, destEnvironmentID := range *req.Links {
+			if contains(destEnvironmentID, currentLinkedEnvIDs) {
+				continue
+			}
+
+			fqDestEnvID := addLayer0Prefix(e.Config.Instance(), destEnvironmentID)
+			sg, err := readSG(e.AWS.EC2, getEnvironmentSGName(fqDestEnvID))
+
+			if err != nil {
+				return checkForAWSError(err)
+			}
+
+			sourceSGID := aws.StringValue(sourceEnvSG.GroupId)
+			destSGID := aws.StringValue(sg.GroupId)
+
+			if err := e.createIngressInput(sourceSGID, destSGID); err != nil {
+				return err
+			}
+
+			if err := e.setLinkTags(req.EnvironmentID, *req.Links); err != nil {
+				return err
+			}
+		}
+
+		// remove - check if links in request are missing from current links
+		for _, destEnvironmentID := range currentLinkedEnvIDs {
+			if contains(destEnvironmentID, *req.Links) {
+				continue
+			}
+
+			fqDestEnvID := addLayer0Prefix(e.Config.Instance(), destEnvironmentID)
+			sg, err := readSG(e.AWS.EC2, getEnvironmentSGName(fqDestEnvID))
+
+			if err != nil {
+				return checkForAWSError(err)
```

Comment:
again, don't want to return here if err is nil. Probably would be best to not use a helper for this.

Reply:
Ok - I think I was only thinking of a single link being deleted but in the case of multiple links being deleted this should be a continue. Moved out of helper and updated to continue ✅